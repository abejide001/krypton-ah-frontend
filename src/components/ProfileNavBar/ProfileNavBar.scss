.centered.middle.aligned.column.row.profileNavBar {
  padding: 1rem 10rem;
  background-color: #c4c4c4;
  &.five {
    padding: 1em 8rem;
  }
  .profileNavBarBars {
    display: none;
  }
}
@media (max-width: 767px) {
  .centered.middle.aligned.column.row.profileNavBar {
    padding: 1.2rem 0rem;
    & .desktop {
      display: none;
    }
    .profileNavBarBars {
      display: block;
      position: relative;

      i.big.icon {
        float: left;
        margin-left: 5%;
      }
    }
    .accordionContent {
      position: absolute;
      background-color: rgb(236, 235, 235);
      z-index: 1;
      top: 3.7em;
    }
    .profileMenu {
      flex-direction: row;
      justify-content: flex-start;
      padding: 4% 6%;
      .profileMenuIcon {
        font-size: 1rem;
        width: 1.7em;
      }
      .profileMenuText {
        display: inline;
        font-size: 1rem;
        margin-left: 3em;
      }
    }
  }
}
