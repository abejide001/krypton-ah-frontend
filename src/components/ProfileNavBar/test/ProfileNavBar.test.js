import React from 'react';
import { shallow } from 'enzyme';
import ProfileNavBar from '..';

describe('ProfileNavBar', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<ProfileNavBar />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<ProfileNavBar user="owner" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot', () => {
    const wrapper = shallow(<ProfileNavBar user="authenticated" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should toggle active state', () => {
    const wrapper = shallow(<ProfileNavBar />);
    wrapper.instance().toggleMenu();
    expect(wrapper.state().active).toEqual(true);
  });
});
