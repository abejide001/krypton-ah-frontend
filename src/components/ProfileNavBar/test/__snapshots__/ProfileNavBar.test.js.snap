// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileNavBar should match snapshot 1`] = `
<GridRow
  centered={true}
  className="profileNavBar"
  columns={4}
  verticalAlign="middle"
>
  <Accordion
    className="profileNavBarBars"
  >
    <AccordionTitle
      active={false}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="bars"
        size="big"
      />
    </AccordionTitle>
    <AccordionContent
      active={false}
      className="accordionContent"
    >
      <GridColumn
        className="mobile"
        key="PROFILE"
      >
        <ProfileMenu
          count={null}
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="PROFILE"
          to="/profile"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="ARTICLES"
      >
        <ProfileMenu
          count="6"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="ARTICLES"
          to="/user-articles"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWERS"
      >
        <ProfileMenu
          count="10"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWERS"
          to="/followers"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWING"
      >
        <ProfileMenu
          count="12"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWING"
          to="/following"
        />
      </GridColumn>
    </AccordionContent>
  </Accordion>
  <GridColumn
    className="desktop"
    key="PROFILE"
  >
    <ProfileMenu
      count={null}
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="PROFILE"
      to="/profile"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="ARTICLES"
  >
    <ProfileMenu
      count="6"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="ARTICLES"
      to="/user-articles"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWERS"
  >
    <ProfileMenu
      count="10"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWERS"
      to="/followers"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWING"
  >
    <ProfileMenu
      count="12"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWING"
      to="/following"
    />
  </GridColumn>
</GridRow>
`;

exports[`ProfileNavBar should match snapshot 2`] = `
<GridRow
  centered={true}
  className="profileNavBar"
  columns={6}
  verticalAlign="middle"
>
  <Accordion
    className="profileNavBarBars"
  >
    <AccordionTitle
      active={false}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="bars"
        size="big"
      />
    </AccordionTitle>
    <AccordionContent
      active={false}
      className="accordionContent"
    >
      <GridColumn
        className="mobile"
        key="PROFILE"
      >
        <ProfileMenu
          count={null}
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="PROFILE"
          to="/profile"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="BOOKMARKS"
      >
        <ProfileMenu
          count={null}
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="BOOKMARKS"
          to="/bookmarks"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="ARTICLES"
      >
        <ProfileMenu
          count="6"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="ARTICLES"
          to="/user-articles"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWERS"
      >
        <ProfileMenu
          count="10"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWERS"
          to="/followers"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWING"
      >
        <ProfileMenu
          count="12"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWING"
          to="/following"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="SETTINGS"
      >
        <ProfileMenu
          count={null}
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="SETTINGS"
          to="/settings"
        />
      </GridColumn>
    </AccordionContent>
  </Accordion>
  <GridColumn
    className="desktop"
    key="PROFILE"
  >
    <ProfileMenu
      count={null}
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="PROFILE"
      to="/profile"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="BOOKMARKS"
  >
    <ProfileMenu
      count={null}
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="BOOKMARKS"
      to="/bookmarks"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="ARTICLES"
  >
    <ProfileMenu
      count="6"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="ARTICLES"
      to="/user-articles"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWERS"
  >
    <ProfileMenu
      count="10"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWERS"
      to="/followers"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWING"
  >
    <ProfileMenu
      count="12"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWING"
      to="/following"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="SETTINGS"
  >
    <ProfileMenu
      count={null}
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="SETTINGS"
      to="/settings"
    />
  </GridColumn>
</GridRow>
`;

exports[`ProfileNavBar should match snapshot 3`] = `
<GridRow
  centered={true}
  className="profileNavBar"
  columns={5}
  verticalAlign="middle"
>
  <Accordion
    className="profileNavBarBars"
  >
    <AccordionTitle
      active={false}
      onClick={[Function]}
    >
      <Icon
        as="i"
        name="bars"
        size="big"
      />
    </AccordionTitle>
    <AccordionContent
      active={false}
      className="accordionContent"
    >
      <GridColumn
        className="mobile"
        key="PROFILE"
      >
        <ProfileMenu
          count={null}
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="PROFILE"
          to="/profile"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="ARTICLES"
      >
        <ProfileMenu
          count="6"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="ARTICLES"
          to="/user-articles"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWERS"
      >
        <ProfileMenu
          count="10"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWERS"
          to="/followers"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
        key="FOLLOWING"
      >
        <ProfileMenu
          count="12"
          menuItem={
            <Image
              as="img"
              className="profileMenuIcon"
              src="test-file-stub"
              ui={true}
            />
          }
          text="FOLLOWING"
          to="/following"
        />
      </GridColumn>
      <GridColumn
        className="mobile"
      >
        <ProfileMenu
          count=""
          menuItem={
            <Button
              as="button"
              secondary={true}
            >
              Follow
            </Button>
          }
          text=""
          to=""
        />
      </GridColumn>
    </AccordionContent>
  </Accordion>
  <GridColumn
    className="desktop"
    key="PROFILE"
  >
    <ProfileMenu
      count={null}
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="PROFILE"
      to="/profile"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="ARTICLES"
  >
    <ProfileMenu
      count="6"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="ARTICLES"
      to="/user-articles"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWERS"
  >
    <ProfileMenu
      count="10"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWERS"
      to="/followers"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
    key="FOLLOWING"
  >
    <ProfileMenu
      count="12"
      menuItem={
        <Image
          as="img"
          className="profileMenuIcon"
          src="test-file-stub"
          ui={true}
        />
      }
      text="FOLLOWING"
      to="/following"
    />
  </GridColumn>
  <GridColumn
    className="desktop"
  >
    <ProfileMenu
      count=""
      menuItem={
        <Button
          as="button"
          secondary={true}
        >
          Follow
        </Button>
      }
      text=""
      to=""
    />
  </GridColumn>
</GridRow>
`;
